### PR TITLE
Add GitHub Actions for Bitcoin, c-lightning, and LND setup and Integration Tests

### PR DESCRIPTION
**Pull request by lnd3v**
- Added a new workflow for running integration tests in .github/workflows/integration_tests.yaml.
- Implemented new GitHub Actions for setting up Bitcoin, c-lightning, and LND environments in .github/actions/setup-bitcoin/action.yaml, .github/actions/setup-clightning/action.yaml, and .github/actions/setup-lnd/action.yaml respectively.
- Updated .gitignore.
- Currently continue-on-error=true until all tests are fixed.

### CODE REVIEW

File: .github/actions/setup-lnd/action.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    required: true
+    default: 'v0.16.2-breez'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.16
```
Comment (JssDWt) [inline .github/actions/setup-lnd/action.yaml:21]: minor thing: go version should be 1.19

**Review by JssDWt — COMMENTED** [review]
I think this flow is equivalent to `testOpenZeroConfSingleHtlc`. But that one has some assertions and a delay to ensure there is no race starting the LSPD interceptor. So this one we can probably do without? What do you think?

File: .github/workflows/integration_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          go build -o lspd_plugin ./cln_plugin/cmd
+        shell: bash
+
+      - name: Test LSPD
+        run: |
+          go get github.com/breez/lspd/itest
+          TESTRE="TestLspd"
+          SKIP_TESTOFFLINENOTIFZEROCONF=1 \
+          SKIP_TESTZERORESERVE=1 \
+          go test -timeout 20m -v \
```
Comment (JssDWt) [inline .github/workflows/integration_tests.yaml:49]: Let's make this 45m or something, to have some legroom here.

**Comment by lnd3v** [comment]
Duplicate test removed.

**Comment by JssDWt** [comment]
I took the liberty of squashing some things together and making the tests run in a matrix.

**Pull request merged**